core/src/main/java/org/apache/brooklyn/core/entity/AbstractApplication.java

```diff
@@ -148,19 +149,23 @@ public void start(Collection<? extends Location> locations) {
         // 2016-01: only pass locations passed to us, as per ML discussion
         Collection<? extends Location> locationsToUse = locations==null ? ImmutableSet.<Location>of() : locations;
         ServiceProblemsLogic.clearProblemsIndicator(this, START);
-        ServiceStateLogic.ServiceNotUpLogic.updateNotUpIndicator(this, Attributes.SERVICE_STATE_ACTUAL, "Application starting");
+        ServiceStateLogic.ServiceNotUpLogic.updateNotUpIndicator(this, Attributes.SERVICE_STATE_ACTUAL, Lifecycle.STARTING);
         setExpectedStateAndRecordLifecycleEvent(Lifecycle.STARTING);
         try {
             preStart(locationsToUse);
-            // if there are other items which should block service_up, they should be done in preStart
-            ServiceStateLogic.ServiceNotUpLogic.clearNotUpIndicator(this, Attributes.SERVICE_STATE_ACTUAL);
-            
+
+            // Opportunity to block startup until other dependent components are available
+            Object val = config().get(START_LATCH);
+            if (val != null) log.debug("{} finished waiting for start-latch; continuing...", this, val);
```

Comment:
I see you've copy-pasted the log message from `MachineLifecycleEffectorTasks`. I don't think the `val` will be interesting (presumably it will simply be "true"). Worth tidying up this log statement and also those in `MachineLifecycleEffectorTasks`.